core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -1077,9 +1077,11 @@ public void disconnect(String reason) {
         if (!closed) {
             loggedIn = false;
 
-            // Fire SessionDisconnectEvent
             SessionDisconnectEvent disconnectEvent = new SessionDisconnectEvent(this, reason);
-            geyser.getEventBus().fire(disconnectEvent);
+            if (authData != null && clientData != null) {
+                // Fire SessionDisconnectEvent
```

Comment:
i would just explain the case(s) of the datas being null (invalidly joining, or if there is anything else. just going off what you were saying in the description)

Reply:
sure, I'll add a comment in a sec